src/components/table/table.tsx

```diff
@@ -1,47 +1,85 @@
 import './table.css'
 
-import Box from '@mui/material/Box'
+import { useMediaQuery, useTheme } from '@mui/material'
+import Stack from '@mui/material/Stack'
 import { default as MuiTable } from '@mui/material/Table'
 import TableBody from '@mui/material/TableBody'
 import TableCell from '@mui/material/TableCell'
 import TableContainer from '@mui/material/TableContainer'
 import TableHead from '@mui/material/TableHead'
+import TablePagination from '@mui/material/TablePagination'
 import TableRow from '@mui/material/TableRow'
 import Typography from '@mui/material/Typography'
-import React from 'react'
+import React, { useState } from 'react'
 
 import { capitalizeFirstLetter } from '../helper'
 import type { TableData, TableHeader } from '../types'
 
+const paginationThreshold = 10
+
 export default function Table(props: TableData) {
+  const isDataGreaterThanThreshold = props.data.length > paginationThreshold
+
+  const [rowsPerPage, setRowsPerPage] = useState(
+    isDataGreaterThanThreshold ? paginationThreshold : -1
+  )
+  const [currentPage, setCurrentPage] = useState(0)
+
   if (props.data.length === 0) {
     return <Typography variant="body2">No data available</Typography>
   }
 
+  const theme = useTheme()
+  const isMobile = useMediaQuery(theme.breakpoints.down('md'))
+
   const dataKeys = Array.from(
     new Set(props.data.flatMap((rowData) => Object.keys(rowData)))
   )
   const headers: TableHeader[] =
     props.headers || dataKeys.map((key) => ({ dataKey: key }))
 
+  const dataToDisplay =
+    rowsPerPage > 0
+      ? props.data.slice(
+          currentPage * rowsPerPage,
+          currentPage * rowsPerPage + rowsPerPage
+        )
+      : props.data
+
+  function handleChangePage(
```

Comment:
I wonder if you want to remove this function because it's only one line of code.